### DISCUSSION TITLE
Social Media Login using Rest API

### DISCUSSION THREAD

**Discussion by nithin-pb** [supabase/supabase#2647] [Questions] [answered]
Hi,
I am trying to create google/Facebook authentication to my application. right now I am using the superbase rest API endpoint for authentication. How to integrate social media login using default endpoints?

**Comment by inian** [comment, marked as answer]
Hi @nithinpachday, we have guides for different OAuth providers in our website. This is the one for Facebook - https://supabase.io/docs/guides/auth/auth-facebook and a video explaining it here - https://www.youtube.com/watch?v=EbV746pWDas

Let me know if there is something thats not clear in these guides and we will improve it.

**Reply by nithin-pb** [reply]
Hi @inian,

Thanks for this. I would like to know if there is any way that I can implement the same using built-in REST APIs provided by superbase. the above tutorials are mentioning about connecting via js

**Reply by inian** [reply]
The API docs for our Auth server are here - https://github.com/supabase/gotrue and you can see how our client library interacts with the API here - https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts

It would be a lot more work using the API directly since you would need to handle the session, handle the redirect URLs, etc yourself.

**Reply by nithin-pb** [reply]
Thanks, @inian. I decided to use supabase-js lib. I would like to know one more thing. Using superbase-js I was able log in with third-party auth providers, however, how should I request a new access token? should I use the superbase-refresh-token method?

**Reply by inian** [reply]
Yup you can get a new token using that method Nithin https://supabase.github.io/gotrue-js/classes/gotrueapi.html#refreshaccesstoken
